ProcessLib/ThermoMechanicalPhaseField/ThermoMechanicalPhaseFieldFEM-impl.h

```diff
@@ -0,0 +1,644 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *  \file   ThermoMechanicalPhaseFieldFEM-impl.h
+ *  Created on January 8, 2018, 3:00 PM
+ */
+#pragma once
+
+#include "ThermoMechanicalPhaseFieldFEM.h"
+
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "ProcessLib/CoupledSolutionsForStaggeredScheme.h"
+
+namespace ProcessLib
+{
+namespace ThermoMechanicalPhaseField
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void ThermoMechanicalPhaseFieldLocalAssembler<ShapeFunction, IntegrationMethod,
+                                DisplacementDim>::
+    assembleWithJacobian(double const t, std::vector<double> const& local_x,
+                         std::vector<double> const& local_xdot,
+                         const double /*dxdot_dx*/, const double /*dx_dx*/,
+                         std::vector<double>& /*local_M_data*/,
+                         std::vector<double>& /*local_K_data*/,
+                         std::vector<double>& local_rhs_data,
+                         std::vector<double>& local_Jac_data)
+{
+    using RhsVector =
+        typename ShapeMatricesType::template VectorType<phasefield_size +
```

Comment:
If this function is for the monolithic scheme of phase-field thermomechanics, the size of  RhsVector  or JacobianMatrix  should be `temperature_size + phasefield_size + displacement_size`.

Reply:
If only the staggered scheme is needed, you can remove this function.

Reply:
this function is removed as well as the monolithic scheme.